packages/protocol/contracts/L2/CrossChainOwned.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity 0.8.20;
+
+import "lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+
+import "../common/EssentialContract.sol";
+import "../signal/ISignalService.sol";
+
+/// @title CrossChainOwned
+/// @notice This contract's owner lives on another chain who uses signal for transaction approval.
+abstract contract CrossChainOwned is EssentialContract {
+    uint64 public ownerChainId; // slot 1
+    uint64 public nextTxId;
+    uint256[49] private __gap;
+
+    event TransactionExecuted(uint64 indexed txId, bytes32 indexed approvalHash);
+
+    error INVALID_PARAMS();
+    error INVALID_EXECUTOR();
+    error TX_NOT_APPROVED();
+    error TX_REVERTED();
+    error NOT_CALLABLE();
+
+    function executeApprovedTransaction(
```

Comment:
Why does this function check its own proof instead of doing things similarly to  what `checkProcessMessageContext` where it's called directly by the bridge contract and just the original sender on the source chain needs to be checked?

Reply:
This new contract doesn't use the Bridge, instead, it interacts with the SignalService directly on the source chain directly.

Reply:
Any reason for that? Seems like this type of thing will be a common thing that people have to do for these cross layer transactions. It feels a bit strange that even we wouldn't use the standard bridge to do this.

Reply:
Using the SignalService is intuitive to me, but we can also use the Bridge. By using the bridge, maybe we can avoid the introduction of the new CrossChainOwned contract. I'll give it a try and then lets compare the two solutions.